src/org.hdfgroup.object/hdf/object/CompoundDS.java

```diff
@@ -97,29 +97,6 @@
      */
     protected transient Object[] memberDims;
 
-    /**
-     * The datatypes of compound members.
-     */
-    protected Datatype[] memberTypes;
```

Comment:
Everything removed from this file definitely belongs back here and not in Dataset.java. memberTypes, isMemberSelected, getMemberCount, etc. only apply to CompoundDS datasets, not general Datasets.